packages/fuel-indexer-tests/components/indices/fuel-indexer-test/src/lib.rs

```diff
@@ -705,4 +705,35 @@ mod fuel_indexer_test {
 
         e.save();
     }
+
+    fn fuel_indexer_trigger_union_type(_b: BlockData) {
+        Logger::info(
+            "fuel_indexer_trigger_union_type handling trigger_union_type event.",
+        );
+
+        let unindexable = UnIndexableUnionEntity {
+            a: Some(2),
+            b: None,
+            c: Some(6),
+            union_type: Some(UnionType::B.into()),
+        };
+
+        let unindexable_container = UnIndexableUnionContainerEntity {
+            id: 1,
+            union_entity: Some(unindexable.into()),
+            union_type: UnionType::B.into(),
+        };
+
+        unindexable_container.save();
+
+        let indexable = IndexableUnionEntity {
+            id: 1,
+            a: Some(5),
+            b: Some(10),
+            c: None,
+            union_type: Some(UnionType::A.into()),
+        };
+
+        indexable.save();
+    }
```

Comment:
Regarding devx:

I'm thinking of ways that we could make things a bit easier on the indexer dev here. Is it possible that we can represent these union types with Rust enums in which each variant is associated with a corresponding struct?

For example:

```rust
let variant_a = IndexableUnionA {
    id: 123,
    a: 456,
};

let indexable_union = IndexableUnion::A(variant_a);
indexable_union.save();
```

This would require us to add functionality to the schema macro to create an `enum` for each GraphQL `union` type, but that should be fairly straightforward as we could just name each of the variant arms after the entity types themselves.

Reply:
See here for more detail on why this approach https://github.com/FuelLabs/fuel-indexer/issues/988#issuecomment-1584518964